testing/e2e/init.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Configure git
+git config --global user.email "wix_build_tools@wix.com"
+git config --global user.name "Wix"
+
+if [[ -z "${TEST_TMPDIR:-}" ]]; then
+  export TEST_TMPDIR="$(mktemp -d ${TMPDIR:-/tmp}/bazel-test.XXXXXXXX)"
+fi
+
+if [[ ! -e "${TEST_TMPDIR}" ]]; then
+  mkdir -p -m 0700 "${TEST_TMPDIR}"
+fi
+
+# Name of current test
+TEST_name=""
+
+# All tests log file
+TEST_log=$TEST_TMPDIR/log
+
+# Cleanup test environment upon exit
+trap 'cleanup' EXIT
+
+function cleanup() {
+  # Clean TEST_TMPDIR and verify path contains an identifier before deletion
+  if [[ -d ${TEST_TMPDIR} && "${TEST_TMPDIR}" == *"bazel-test"* ]]; then
+    rm -rf ${TEST_TMPDIR}
+    log_info "Cleaned local test environment"
+  else
+    log_info "Test environment was cleaned already !"
+  fi
+
+  delete_git_cached_directory
```

Comment:
It is the same behavior as you'll probably never instantiate a random file from a testing framework, rather, you'll read the docs or find by yourself the entry point of the testing framework.

Remember that we're talking about bash scripts.. it's not something structured like Scala/Java, and because of that, there are assumption such as which file(s) to run, execution path etc.. which have to be enforced.

Regardless, I've sourced relevant sh files for all custom defined functions - it'll be easier to figure out the function origin 👍🏽